### DISCUSSION TITLE
Copilot now supports various models (GPT, Anthropic, and Gemini). How will the copilot adapter handle this?

### DISCUSSION THREAD

**Discussion by PyDataBlog** [olimorris/codecompanion.nvim#387] [Q&A] [answered]
In VS Code copilot chat, users select which model and chat proceeds. How will the copilot adapter handle this selection?

<img width="424" alt="Screenshot 2024-11-03 at 10 43 36" src="https://github.com/user-attachments/assets/726d4c6a-c302-4f7e-9c7c-fcd6abe58975">

**Comment by olimorris** [comment]
By you making a PR? We have a rather well fleshed out guide to Adapters, [here](https://github.com/olimorris/codecompanion.nvim/blob/main/doc/ADAPTERS.md)

**Comment by olimorris** [comment]
Also, if someone can give me the model names and parameters it's a very simple addition

**Reply by PyDataBlog** [reply]
> Also, if someone can give me the model names and parameters it's a very simple addition

I'm trying to actually find the model names

**Reply by PyDataBlog** [reply]
> Also, if someone can give me the model names and parameters it's a very simple addition

Here's what I have found so far:

```lua
'gpt4-o',
'o1-preview',
'o1-mini',
```

Can't find anything yet on Gemini and Anthropic models

**Reply by olimorris** [reply]
I had a search on Thursday and couldn't find anything. I [see](https://github.com/zed-industries/zed/blob/main/crates/copilot/src/copilot_chat.rs) Zed haven't updated their implementation yet either

**Reply by PyDataBlog** [reply]
> I had a search on Thursday and couldn't find anything. I [see](https://github.com/zed-industries/zed/blob/main/crates/copilot/src/copilot_chat.rs) Zed haven't updated their implementation yet either

tbh it just dropped  out of the oven so very bleeding edge as it stands. Guess we just need to be ready. Will raise an issue there to see if they can do the heavy lifting there ;)

**Reply by PyDataBlog** [reply]
> I had a search on Thursday and couldn't find anything. I [see](https://github.com/zed-industries/zed/blob/main/crates/copilot/src/copilot_chat.rs) Zed haven't updated their implementation yet either

Issue raised on Zed: https://github.com/zed-industries/zed/issues/20133

**Reply by PyDataBlog** [reply]
> I had a search on Thursday and couldn't find anything. I [see](https://github.com/zed-industries/zed/blob/main/crates/copilot/src/copilot_chat.rs) Zed haven't updated their implementation yet either

Here's the current Zed PR implementing more models:

https://github.com/zed-industries/zed/pull/19968

**Reply by olimorris** [reply]
The Claude model drops right in and works perfectly for me. The o1 may need some tweaking in regards to streaming.

Seems that for CodeCompanion's purposes, the API schema remains the same. So GitHub must have done some work to transform the OpenAI endpoint to Anthropic.

I'll push this update later today after some testing. I'll be dropping version 9.0 of the plugin later today with some huge updates.

**Reply by PyDataBlog** [reply]
> The Claude model drops right in and works perfectly for me. The o1 may need some tweaking in regards to streaming.
> 
> 
> 
> Seems that for CodeCompanion's purposes, the API schema remains the same. So GitHub must have done some work to transform the OpenAI endpoint to Anthropic.
> 
> 
> 
> I'll push this update later today after some testing. I'll be dropping version 9.0 of the plugin later today with some huge updates.

That's pretty much what most want so I'm looking forward to using it tonight 🙂

**Reply by olimorris** [reply]
In the meantime you can just extend the copilot adapter and add the model as per the README.

Performance seems great so far

**Reply by PyDataBlog** [reply]
> In the meantime you can just extend the copilot adapter and add the model as per the README.
> 
> Performance seems great so far

Claude is the best for coding imo so that'll be my default. Looking forward to the update tonight

**Reply by PyDataBlog** [reply]
> In the meantime you can just extend the copilot adapter and add the model as per the README.
> 
> Performance seems great so far

Release today expected? If not how do I extend the adapter for Claude?

**Reply by olimorris** [reply]
As with all features, everything is documented in the [README](https://github.com/olimorris/codecompanion.nvim#electric_plug-adapters). Search for "Changing an Adapter's Default Model".

**Reply by adhi-thirumala** [reply]
I don't currently see this in the readme. what specifically do I put in the config to switch to claude on copilot?
`adapters = {
    copilot = function()
      return require("codecompanion.adapters").extend("copilot", {
        schema = {
          model = {
            default = "claude-3.5-sonnet",
          },
        },
      })
    end,
  }`
this seems to not work

**Comment by olimorris** [comment, marked as answer]
Hi All,

Update on this, I added support for o1 models last week. Both OpenAI and Copilot can utilise them. As you're probably aware, they don't offer a streaming endpoint so you'll be waiting a while to see the response.

**Reply by PyDataBlog** [reply]
> Hi All,
> 
> 
> 
> Update on this, I added support for o1 models last week. Both OpenAI and Copilot can utilise them. As you're probably aware, they don't offer a streaming endpoint so you'll be waiting a while to see the response.

It seems streaming support is already available?

https://x.com/openaidevs/status/1858609150999359559?s=46&t=LgoR4FET5B7oorqnGbecPQ

**Reply by olimorris** [reply]
Yeah looks like they added that less than 24 hours ago. Great news and thanks for sharing.

Doesn't seem to be enabled for Copilot yet.
